addons/rivet/devtools/rivet_cli_output.gd

```diff
@@ -11,18 +11,36 @@ enum ExitCode {
 	# TODO: fill with the rest of the exit codes
 }
 
-func _init(exit_code: int, output: Array) -> void:
+func _init(exit_code: int, internal_output: Array) -> void:
 	self.exit_code = exit_code
 
+	if internal_output and not internal_output.is_empty():
+		_parse_output(internal_output)
+
+func _parse_output(internal_output: Array) -> void:
+	self.formatted_output = internal_output.map(
+		func (line: String): 
+			# TODO(compat): test this on windows as it may have different EOFs
+			return line.split("\n", false)
+	).reduce(
+		func (accum, line): 
+			accum.append_array(line) 
+			return accum
+	,[])
+
+	var lines_with_json = self.formatted_output.filter(
+		func (line: String): 
+			return line.find("{") != -1
+	)
+
+	if lines_with_json.is_empty():
+		print("No JSON output found")
+		return
+
+	var line_with_json: String = lines_with_json.front()
```

Comment:
It looks like this code reads multiple lines from the CLI instead of just one. Are you going on the assumption that multiple lines might come from the CLI? Or are there any cases you've come across where this happens?

Reply:
I'm happy to leave this in, I'm not totally convinced that the CLI will behave 100% just yet :P